### ISSUE TITLE
Project 2 Peer Review thread

### ISSUE THREAD

**Issue by susanBuck** [susanBuck/e28-spring21#36]
In this week’s upcoming videos (Week 12, Tue Apr 20), I'll cover the instructions for completing a peer review of Project 2. You will share your completed review in a reply to this thread.

+ Full instructions of the peer review process will be covered in the Week 12 assignment.
+ [Peer review spreadsheet...](https://docs.google.com/spreadsheets/d/1ixtbHS8XuHpeQtsSkvpDMqLRo8UEc4sp0Sf1cmPnUUk/edit#gid=0)

**Comment by susanBuck** [comment]
This is an example...

**Comment by keamarg** [comment]
### What is the Github URL of the project you are reviewing?
[Vue of the Yard](http://e28p2.vueoftheyard.com/
)
### Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?
+ No, all good.

### Are there any parts of the code that you found interesting or taught you something new?
+ In my own project, I completely forgot that you can bind props to the routes. After looking at the code, I updated my own project to reflect this (although it is not really integral to the experience as is the case with this project).
+ I liked the [validateForm](https://github.com/bryonjustice/e28/blob/07606731b0d5ccd38e723355ceb60cf9d21d7b95/p2/src/components/pages/PostAddPage.vue#L183) function. I will probably add a similar approach to client side validation in project 3 
### Are there any parts of the code that you don't understand?
+ In my own version of [app.js](https://github.com/bryonjustice/e28/blob/07606731b0d5ccd38e723355ceb60cf9d21d7b95/p2/src/common/app.js#L2) I changed the url in base to reflect the production server, but this does not seem to be necessary. I know this was addressed in one of the videos, but I am in doubt how the baseURL: process.env.VUE_APP_API_URL ?? [url] actually provides a url to axios.
+ Apart from that, I think it is all pretty clear, and follows the line of instruction we have had very well.

### Are there any parts of the code that you feel could be written in a more efficient or clearer manner?
+ I think that the code in general is very neat and well written. If I had to nitpick, I guess consistency is always good. For instance, if there is a v-if in and element, I try to always put it in the beginning of the tag rather than at the end like [here](https://github.com/bryonjustice/e28/blob/07606731b0d5ccd38e723355ceb60cf9d21d7b95/p2/src/components/pages/PostAddPage.vue#L3). I think this is especially important with the v-if, because it determines whether or not the rest matters, but to be honest, I probably do the same myself sometimes.

### Are there aspects of the code that you feel were not self-evident and would benefit from comments?
+ Maybe the "[ShowPost"](https://github.com/bryonjustice/e28/blob/07606731b0d5ccd38e723355ceb60cf9d21d7b95/p2/src/components/ShowPost.vue#L2) component could benefit from a small comment. Especially this part where the id is sliced (and why):

```
      <div v-else class="card">
            <img v-bind:src="imgSrc" />
            <div class="blurb"> 
                <span class="blurb">{{ post.cover_blurb }}</span><br/>
                {{ post.published_date }}<br/>
            </div>
            <div class="id">
                {{ (('0'+post.id).slice(-2)) }}<br/>
            </div>
            <div class="intro">
                {{ post.intro }}<br/>
            </div>
        </div>
```

### Are there any best practices discussed in course material that you feel were not addressed in the code?
+ No, I cannot think of anything.

### Do you have any additional comments not covered in the above questions?
+ Nice project:)

**Comment by misskimdo** [comment]
**What is the Github URL of the project you are reviewing?**

[Jimbo's Recipes](http://e28p2.sean-bollin.com/)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 

No errors or warnings.

**Are there any parts of the code that you found interesting or taught you something new?**

- I didn't use the [name option within export default](https://github.com/seanbollin/e28/blob/20e38b7744508cdc2c98dbca22f5b2537eb3d355/p2/src/components/pages/HomePage.vue#L10) and looked into this. Learned it's good for debugging. Is it good to include this as best practices? I only saw it used on the homepage.
- Your method for adding to a favorites list is simple and clear-cut. This would be a great feature in my project as well.


**Are there any parts of the code that you don't understand?**

- I wasn't sure why some pages had [```v-bind:detailed```](https://github.com/seanbollin/e28/blob/20e38b7744508cdc2c98dbca22f5b2537eb3d355/p2/src/components/pages/RecipesPage.vue#L7) when there weren't "detailed" portions of the recipes? Not sure if I missed something?
- Similary with [```inheritAttrs: false,```](https://github.com/seanbollin/e28/blob/20e38b7744508cdc2c98dbca22f5b2537eb3d355/p2/src/components/pages/RecipesPage.vue#L19). Not sure where this was being used.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

If the above items were not meant to be in the code, I think cleaning those items out would suffice. Everything else is clear.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

No, I think the code was easy to read.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

I think after clicking "Add to Favorites," it would've been nice to indicate that to the user via some form of feedback. E.g. fade-in-out star next to the item

**Do you have any additional comments not covered in the above questions?**

I think your project was clean and to the point. As someone who has lived in Louisiana, I enjoyed the swamp vernacular. ;)

**Comment by millierni** [comment]
**What is the Github URL of the project you are reviewing?**

[acomjean](https://github.com/acomjean/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

+ There are no errors or warnings in the console when I'm using the website.
+ There is a title error in [`index.html`](https://github.com/acomjean/e28/blob/226b4432fcf6f7a3781d34cff67577711d704471/p2/public/index.html#L8).  
It shows some html code instead of plain text.

```html
<title><E28 P2></title>
```

Need to be changed to:

```html
<title>E28 P2</title>
```

+ In the Itinerary page, we can add an empty comment on the artworks.  
Maybe, add an `if statement` to reject empty comment. 

**Are there any parts of the code that you found interesting or taught you something new?**

+ No, but I find your code well written in general.

**Are there any parts of the code that you don't understand?**

+ I don't fully understand why you bind to your components [`ArtistGrid`](https://github.com/acomjean/e28/blob/226b4432fcf6f7a3781d34cff67577711d704471/p2/src/components/pages/ArtistListPage.vue#L174-L182), the same data that are already bind in [`App.vue`](https://github.com/acomjean/e28/blob/226b4432fcf6f7a3781d34cff67577711d704471/p2/src/App.vue#L35-L45)?  

You can simply import data as props directly from `App.vue`.  
No need to re-bind the data.  
Maybe I'm missing something here.






**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

+ In [`ArtistListPage.vue`](https://github.com/acomjean/e28/blob/226b4432fcf6f7a3781d34cff67577711d704471/p2/src/components/pages/ArtistListPage.vue#L155-L161) I would have structured the code a little bit more clearer, in my opinion.  
Something like this:

```html
<p>
    Total Artists: 
    {{ artistCount }}
    <br />
    Artists In Your Itinerary: 
    {{ itineraryDetails.length}}
    <br />
    Artists in itinerary not visited: 
    {{ artistsToVisit.length}}
    <br />
</p>
```

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

No, the code is clear and well structured.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

+ I found that you are not using axios in [`App.vue`](https://github.com/acomjean/e28/blob/226b4432fcf6f7a3781d34cff67577711d704471/p2/src/App.vue#L49). You can simply import your JSON file [`.../common/all_artists_2021.json`] with axios using the `get` request. Maybe I'm missing something?

**Do you have any additional comments not covered in the above questions?**

+ I suggest you to clean up the tables in your database for your API (`database.sqlite`). There are favorites and products that are not used in this app. P.S. you can used the application interface of SQLite to edit your database in [`.../e28api/core/database/`].
+ BTW, I really like your project idea!

**Comment by bryonjustice** [comment]
Q: What is the Github URL of the project you are reviewing? 
[https://github.com/richiecarey/e28/tree/main/p2]( https://github.com/richiecarey/e28/tree/main/p2  )
 
 Q: Are you able to produce any errors or unexpected results when using the site?  
 I was unable to Add A Task.  
 Steps to Reproduce: 
 1) Navigate to http://e28p2.richiecarey.com/ 
 2) Enter a Task Name in `input#inlineFormInput`. 
 3) Click Add Button.  
 Result:  Task is not added and no feedback is offered to the user.  This blocked testing of showConfirmation.  
 
` TaskUpdateInput.vue` provides a showConfirmation component but does not use the boolean value.  
 
Steps to reproduce:  
1) Navigate to an individual task   Example: http://e28p2.richiecarey.com/task/11. 
2) Enter a field value longer than the expected validator from the API.  
3) Click Update.  
showConfirmation would be false but is not used to conditionally render any content/feedback to the application user.

   Q: Are there any errors or warnings in the console?  
   The Developer Console is clean.  There were no errors or warnings.  
   
   The developer may want to remove console.log messages before checking in code to GitHub. 
   Example: `App.vue`  line 44.   
   
   Q: Are there any parts of the code that you found interesting or taught you something new? 
    The project code is straight forward and correctly applies the learning from the lecture.  I did like the use of the transition effect when tasks were marked done.   I found the application helpful for completing this peer review as I was able to update the names of seeded tasks and use them to annotate my testing as I reviewed the project.     
  
      Q: Are there any parts of the code that you don't understand? 
       The project code is clear and the functionality is easily identifiable.  Components and data are clearly named and relate to the expected functionality.  The order of the source codes and vue options follows the Vue 3 style guide.  The is not any extraneous or unclear code. The components are concise.  The developer is straight forward and correctly applies the learning material from the lecture.     
       
       Q: Are there any parts of the code that you feel could be written in a more efficient or clearer manner?  
       Code was efficient.  I would review the error on adding a task.           
       
       Q: Are there aspects of the code that you feel were not self-evident and would benefit from comments?  
       The project code is straight forward and correctly applies the learning from the lecture.    
       
       Q: Are there any best practices discussed in course material that you feel were not addressed in the code? 
        I would recommend consistently using the vue naming convention for components. Refer to E28 Notes: [https://hesweb.dev/e28/notes/vue/components#naming  ](https://hesweb.dev/e28/notes/vue/components#naming  )
        
        For the most part this convention is followed in the developer’s code.  
     There were a few exceptions. 
      For example reference code in `src/main.js   `  Line: 3   `Import Done from ‘@components/pages/TaskCompletedPage.vue’;`
  
  As a general rule I would encourage using id names that are are specific to the unique HTML element.  
  
  Reference `/src/components/TaskAddInput.vue` line 9.  The input id property is titled `id=“inlineFormInput”.`   
  
  This id reference will work when you only have one input field but would be difficult to discern or remember as the code complexity and number of fields grow.    
  
  Q: Do you have any additional comments not covered in the above questions? 
  Project code is thoughtful.  Code is clean, easy to follow and organized.  The application was correctly built with the `vue-cli-service` build command and production app `/dist` directory was published to GitHub.  The developer has used modern and semantic HTML.   All code follows a consistent spacing standard and indention.   The production application correctly runs from `/dist `and `app.js` models development life cycle support for both production and local.  The repo correctly uses `.gitignore `and does not include unnecessary or secure materials. The relevant API provided included seeders and the resource included validators for `task.json`.  `README.md` follows the provided template and instruction and links were clear.

**Comment by hemanthharvard** [comment]
**What is the Github URL of the project you are reviewing?**

[Hayley's Recipes](https://github.com/hhuang56/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

- No errors or warnings. 👍 

**Are there any parts of the code that you found interesting or taught you something new?**

-     The project code is straight forward and correctly applies the learning from the lecture.

**Are there any parts of the code that you don't understand?**

-     The project code is straight forward.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

- I noticed that in the Contribute tab after submitting a recipe, all the recipes are fetched. But, they are not displayed in that page. Ideally, unnecessary network calls should be avoided.
- Similarly in the Ratings tab after submitting a rating, all the recipes are fetched. It appears that you're keeping the count locally as well and updating the UI instantly. So the latest recipe rating data needs to be fetched only when the page is mounted.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

- I'm not sure why the spread operator was required at [ReviewPage.vue#L36](https://github.com/hhuang56/e28/blob/main/p2/src/components/pages/ReviewPage.vue#L36). Is it because you're modifying the original recipes array with `slice` at [ReviewPage.vue#L28](https://github.com/hhuang56/e28/blob/main/p2/src/components/pages/ReviewPage.vue#L28) and that's causing data mutation?

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

- Maybe the code can be organized better.

**Do you have any additional comments not covered in the above questions?**

- I think your project was clean and to the point. Maybe the CSS styling can be improved. 😃

**Comment by richiecarey** [comment]
**What is the Github URL of the project you are reviewing?**

https://github.com/hemanthharvard/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

Create, Read, Update, and Delete tested with no visible or console errors.

In `main.js` I did notice references to a component that does not exist in the repo, line 13 ...

https://github.com/hemanthharvard/e28/blob/223cb1f078892ed5759f30bc7b8454ca6a498cda/p2/src/main.js#L13

```js
import NoteDetailPage from '@/components/pages/NoteDetailPage.vue';
```

..and lines 29-33 ...

https://github.com/hemanthharvard/e28/blob/223cb1f078892ed5759f30bc7b8454ca6a498cda/p2/src/main.js#L29-L33

```js
{
    path: '/:id',
    component: NoteDetailPage,
    props: true
},
```

**Are there any parts of the code that you found interesting or taught you something new?**

It was interesting to see the [Axios](https://github.com/hemanthharvard/e28/blob/58ec3b292fd65e96603c2e7796ad4efc3a2adc51/p2/src/common/app.js#L1) class in app.js .. makes for a nice, clean way to interact with the API .. well done.

**Are there any parts of the code that you don't understand?**

No, the code is clean, well structured, and easy to follow.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

No, looks good.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

No, not really. The code here, to me, seems self documenting.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

No, it appears to me that naming conventions and other style guide best practices are used consistently.

**Do you have any additional comments not covered in the above questions?**

Nice clean project, meets and exceeds the Project 2 requirements. Well done!

**Comment by seanbollin** [comment]
**What is the Github URL of the project you are reviewing?**

https://github.com/EkaterinaMolchanova/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 

Yes there is a problem - when you refresh the page on, for example, this page: http://e28p2.katemolchanova.me/posts you receive a 404 not found. This is due to the absence of an .htaccess file as specifies in course docs.

**Are there any parts of the code that you found interesting or taught you something new?**

I like the use of [].concat here:

https://github.com/EkaterinaMolchanova/e28/blob/main/p2/src/components/CategoryAndColor.vue#L43

I did not know you could call methods on literal arrays like this.

Also I did not know about justify-content: https://github.com/EkaterinaMolchanova/e28/blob/main/p2/src/components/pages/PostsPage.vue#L64

**Are there any parts of the code that you don't understand?**

I am not totally sure what the "em" means in the CSS here:

https://github.com/EkaterinaMolchanova/e28/blob/main/p2/src/components/pages/AddPostPage.vue#L144

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

No everything looks clean and concise.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

No, the project is pretty small and the code is concise.. I do not believe comments would have helped the code base.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

Yes there was a lack of an .htaccess file so when you refresh the page on a non index page it gives you a 404.

**Do you have any additional comments not covered in the above questions?**

The look and feel of the site is very polished. Great job!

**Comment by acomjean** [comment]
**What is the Github URL of the project you are reviewing?**

https://github.com/EkaterinaMolchanova/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

 - No console Errors.
 - When going to a non-homepage directly we get a not found error (.htaccess file probablyt needs configuration)

**Are there any parts of the code that you found interesting or taught you something new?**

- The array shuffling method. 
- The check for images existing.

**Are there any parts of the code that you don't understand?**

I think the images are manually loaded, though I tried to figure out if it went to the existing site a got them.  I added an existing item from  yout Society6 store it worked except the images. 

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

Very clean and easy to understand.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

None. 

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

None

**Do you have any additional comments not covered in the above questions?**

Nice clean site.  I like the random images on the home page and browsing product to product aspect.

**Comment by hhuang56** [comment]
**What is the Github URL of the project you are reviewing?**
[Bitcoin Mining Calculator](https://github.com/millierni/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
No warnings or errors in the console. However, the the drop down selector for the ASIC Miner on the home page appears to be blank on page load. When clicking on the drop down, it looks like S19 [95th] is the default selection based on the checkmark that appears next to it, but actually clicking on S19 [95th] displays a different set of parameters. Therefore, as the user, I'm somewhat confused about what is actually the default miner being displayed. If it's not any of the miners on the list, maybe create a miner that's called "Default" that can be used on page load instead of a blank selection?

**Are there any parts of the code that you found interesting or taught you something new?**
I have almost no knowledge of how Bitcoin mining works (so if any of my comments missed the mark, that may be why). I had to Google a lot of terms to figure out what sort of information the calculator was conveying, but in the end, it was a cool learning experience for a very cool and helpful tool.

**Are there any parts of the code that you don't understand?**
It was obvious where data was being retrieved from the server, but I couldn't figure out where in the code (or in the UI) data was being persisted to the server (per P2 requirements).

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
Overall, the code was straightforward and clear. However, the code could be even more efficient in BitcoinCalculator.vue.

The following code snippets are repeated several times for the same types of data (hashrate, difficulty, price, etc). You may be able to use v-for to generate these elements against a dictionary. For example, if you have a dictionary, your input elements, table headers, and table rows can use the dictionary to iterate.
`const dictionary = [
{
  title: "Hashrate (TH/s)",
  value: "terahashRate"
}, ...
]`

        <div v-for="item in dictionary" class="col-lg-3">
          <label class="block">
            <h4 class="block-title">item.title</h4> 
            <input type="number" v-model.number="item.value">
          </label>
        </div>

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
No, I was able to read all the code, and I'll trust the author on all the math and calculations!

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
The order of the options could be changed to match best practices. For example, because "computed" was above "data" in the order of options on HomePage.vue, I was stuck for a few seconds trying to figure out where "this.miner" came from.

**Do you have any additional comments not covered in the above questions?**
Just small suggestions - include some instructions on how to use the calculator on the FAQ page and improve the css of the nav bar to match the black/white/grey tones of the rest of the UI.

**Comment by EkaterinaMolchanova** [comment]
**What is the Github URL of the project you are reviewing?** 
https://github.com/kerynegan/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 
I’ve got a mistake “xhr.js:177 GET http://e28api.kerynegan.me/selection 429 (Too Many Requests)” in the console on every visit to the “Matches” page. It did not influence the functionality of the app.

**Are there any parts of the code that you found interesting or taught you something new?** 
There was nothing explicitly new for me. However, I liked high attention to detail in the code, for example splitting and joining items in arrays to look nice for the user for the movie item [here](https://github.com/kerynegan/e28/blob/afe05f8f902d5663b8d7b89153b140eaff41c8d2/p2/src/components/ShowMovie.vue#L9 ) 

**Are there any parts of the code that you don't understand?** 
The code is quite straightforward, clear and well-commented. 

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?** 

From my perspective the code is written in a clear and efficient manner, but I would suggest excluding some lines of the code in main.js and app.js. It could help improve clarity. 

For example, in the main.js router there are paths to individual movi, drink, and meal that are not used in the app. If those will not be required for Project 3, i would suggest deleting them:

```javascript
routes: [ 
  { path: '/', component: HomePage },
  { path: '/movies', component: MoviesPage },
  { path: '/movies/:id', component: MoviesPage, redirect: '/movies' }, 
  { path: '/drinks', component: DrinksPage },
  { path: '/drinks/:id', component: DrinksPage, redirect: '/drinks' }, 
  { path: '/meals', component: MealsPage },
  { path: '/meals/:id', component: MealsPage, redirect: '/meals' }, 
  { path: '/matches', component: MatchesPage },
 ...
  ]
```
In the app.vue I think it’s not required to update the lists of movies, drinks and meals in router-view, as those are not changed in the app. It should be enough to load those on the load of the app:

```javascript
<router-view
  v-bind:userID="userID"
  v-bind:movies="movies"
  v-on:update-movies="loadMovies" 
  v-bind:drinks="drinks"
  v-on:update-drinks="loadDrinks"
  v-bind:meals="meals"
  v-on:update-meals="loadMeals"
  v-bind:selections="selections"
  ...
  ></router-view>
```

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?** 
The code is very well commented.

**Are there any best practices discussed in course material that you feel were not addressed in the code?** 
I’ve not found anything.

**Do you have any additional comments not covered in the above questions?**
From a user perspective I would prefer not to be offered the same items for select/reject the second time, and get the list of items to review and change a decision somewhere in a separate place.

The idea and the app are great! I really enjoyed it and curious to see it as a project 3, as some features requiring logging in are not implemented yet.

**Comment by trevjburns** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/jaredmimms/e28/tree/main/p2](https://github.com/jaredmimms/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 
When adding a new post, submitting the default content seems to work, but whenever I fill out the form with new information, I get an error showing “Content required” and an error adding the post.

**Are there any parts of the code that you found interesting or taught you something new?**
I like that there are many pages for this application!

**Are there any parts of the code that you don't understand?**
No, t all makes sense!

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
After looking through the code, it all seems very clear and concise.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
No, it is very clean!

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
No, the templates and use of data, methods, computed properties, and passing data seem to be written as prescribed in the course material.

**Do you have any additional comments not covered in the above questions?**
Nothing to add, except for nice job!

**Comment by sgreene** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/trevjburns/e28/tree/main/p2](https://github.com/trevjburns/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or
warnings in the console?**

Clicking on an available deck returns an error (404). I believe this is because no route has been setup for the individual items.

**Are there any parts of the code that you found interesting or taught you something new?**

* I did not know about the `default` attribute for props. That's very useful!
* I picked up some css styling tips. It's definitely an area of weakness for me.

**Are there any parts of the code that you don't understand?**
No

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

* Some use of text interpolation is ok, but I think errors should be wrap within an html block and properly styled. Dumping errors to the screen as text won't be pretty.
*  I'm not sure why loadDecks is in [AddNewCard.vue](https://github.com/trevjburns/e28/blob/main/p2/src/components/pages/AddCardPage.vue) and also in the [App.vue](https://github.com/trevjburns/e28/blob/main/p2/src/App.vue). Both files also have the same mounted code. I think this duplication may be unnecessary.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
The duplicate loading of decks, if necessary, could be explained with some comments. Also there may be some dead code like in ShowDecks.vue. If this file is not needed it should be removed. I do not see it as part of the routes so my guess is that it is not needed. 

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

* In [AddCardPage.vue](https://github.com/trevjburns/e28/blob/main/p2/src/components/pages/AddCardPage.vue), the lifecycle event (mounted) should come before `methods`
* This happens for other files as well. See [https://v3.vuejs.org/style-guide/#component-instance-options-order-recommended](https://v3.vuejs.org/style-guide/#component-instance-options-order-recommended) for reference.

**Do you have any additional comments not covered in the above questions?**
I like the simple and clean design. I believe a simple bug fix of adding a new page to the route would fix the 404 issue.

**Comment by jaredmimms** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/seanbollin/e28/tree/main/p2](https://github.com/seanbollin/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
The favorites functionality produces unexpected results. The application allows the user to add the same item to favorites multiple times. 

**Are there any parts of the code that you found interesting or taught you something new?**
I found including a key on a div interesting in lines 4 to 11 of RecipesPage.vue.
`<div v-bind:key="recipe.id" v-for="recipe in recipes">
            <show-recipe
                    v-bind:recipe="recipe"
                    v-bind:detailed="false"
            ></show-recipe>
            <router-link v-bind:to="'/recipe/' + recipe.id">More Details</router-link>
            <button v-on:click="addFavorite(recipe)">Add to Favorites</button>
        </div>`

**Are there any parts of the code that you don't understand?**
I understand all the code.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
The process of adding a favorite should include a check for replicas. 

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
The code did not need more comments. My understanding of Vue.js allowed me to interpret the code without needing comments. 

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
The code did not contain comments so that a person with a beginner's understanding of Vue.js could interpret the code. 

**Do you have any additional comments not covered in the above questions?**
I would include a little more functionality such as a search.

**Comment by kerynegan** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/misskimdo/e28/tree/main/p2](https://github.com/misskimdo/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
I'm not able to view the recipe pages (.../recipe/{id}) for new recipes I added ([example](http://e28p2.kimdo.me/recipe/13)). The console gives me this error:

> TypeError: Cannot read property 'split' of null

Which I think may be caused because the 'Add new recipe' form doesn't have a 'categories' input, but [these lines of code ](https://github.com/misskimdo/e28/blob/80db0c1ef02fbf879ddcaa0e8024a80cd71dc92b/p2/src/components/pages/RecipePage.vue#L6-L15)from @/components/pages/RecipePage.vue are trying to split the value of a null item at each vertical bar.

The recipe does show up [in the API](http://e28api.kimdo.me/), and the previews show in the featured recipes and on the all recipes pages, so it is writing correctly to the server; it just cannot be displayed by the browser.

The other detailed recipe pages (from the seeds) look good though.

**Are there any parts of the code that you found interesting or taught you something new?**
Not new, but I really love the 'show me a random recipe' button. In a future iteration, maybe it would be neat to combine that with a category dropdown, e.g., 'Show me a random \<category\> recipe'?

**Are there any parts of the code that you don't understand?**
I don't think so! The code was very clean.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
No, the code was very clear!

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
There weren't many comments included. I didn't feel as though I understood the code any less because of the lack of comments, but if I didn't have the grounding from the lectures comments would have been useful.

**Are there any best practices discussed in course material that you feel were not addressed?**
I don't think so! Good job!

**Do you have any additional comments not covered in the above questions?**
The README file may need to be updated to add the link to production! I figured out where to go from your p1 README.

Overall I thought this was great! I also really loved the work you did with the branding. The logo for Kim's Kitchen was so cute, and the example image for new recipes made me smile.

**Comment by agt25** [comment]
**What is the Github URL of the project you are reviewing?**

(Kim's Kitchen Recipes) [https://github.com/misskimdo/e28/tree/main/p2] 

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

I found Kim’s app (and Kerynegan’s review) so interesting so I decided to also review it in hopes of debugging the issue. I, too, received: 

> `TypeError: Cannot read property 'split' of null`

I searched for the error on StackOverflow. My undeleted browser history proves that, at one point, I also had the same error! My first inkling was that the ShowRecipe.vue component is not receiving the ‘recipe’ prop correctly and is defaulting to null. But that doesn’t explain how the first few recipes are showing. I was hoping to have been a little more helpful than that! 

I did note that AddRecipe.vue $emits(“update-products”) but App.vue is listening for “update-recipes”. Not sure if that could be the problem because the new recipe’s name is updated on the home page, so a loadRecipes() get request is being made. 

**Are there any parts of the code that you found interesting or taught you something new?**

 I really like how clean and neat the choose() method is on HomePage.vue  (which generates a random recipe)!

**Are there any parts of the code that you don't understand?**

All of the code was pretty clear and easy to read. 

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

Nope, I thought Kim did a great job with her code. 

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

I would’ve loved some comments overall. Kim’s methods are fairly straightforward and readable, but comments always make everything a little better!  


**Are there any best practices discussed in course material that you feel were not addressed?**

Nope! All naming conventions are met, and the code is well designed. 

**Do you have any additional comments not covered in the above questions?**

Awesome job with the branding! It made me feel as If I were sitting outside a seafood restaurant, excitingly awaiting my food whist probably giving the appetizers away to the seagulls. Also, the pomegranate-lychee panda cotta made my stomach growl. My mom loves lychee so thank you for the awesome recipe! I’m excited to try it (by trying it I mean that I’ll be eating it!).

**Comment by katz2row** [comment]
Thanks for sharing! This was an interesting project that could have some possible real-world applications with a few little tweaks.

Honestly, I struggled a great deal with my personal project P2, and so reviewing and evaluating someone else's when I feel like such a mess was quite a struggle. I wish I had more time to balance the demands of this class, my work, and parenthood, but alas, here we are. So please note that is quite possible I didn't fully understand something because of my own confusion and shortcomings and not because of the project itself.

**What is the Github URL of the project you are reviewing?**

<http://e28p2.hesdev.com/>

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 

No, I didn't see any errors or warnings in the console. The one "error" I encountered was a 404 error if I refreshed on any page other than the homepage. I would have to go back to the homepage and refresh and then navigate using the links, but I could not then refresh from that page. I am not sure if this is due to the .htaccess and/or .env.production not being present.

I expected the project to work a little differently, but I believe that was a personal design decision and not necessarily coding issues. The one thing I would have expected is that each quiz go to it's own page. So the Quiz menu item went to whichever quiz you last clicked on, which is ultimately ok as the menu item wouldn't make sense otherwise, but it was unexpected.

**Are there any parts of the code that you found interesting or taught you something new?**

I still feel like I am learning as I go. So it was interesting seeing how someone else set up the data and connected the page contents. 

**Are there any parts of the code that you don't understand?**

I am not sure where or how the QuestionPage.vue is being used as the QuizPage.vue seems to be the one actually being accessed, and the main.js has both pointing to ```/quiz```, but the QuestionPage.vue as an added ```/:id``` and ```props: true```, but I don't see that in action anywhere. 

I also noticed that the ```remodelData()``` function is used when the questions are retrieved from the database, but I don't know what purpose the function actually serves. I guess it could be that it is looking for database changes and rebuilding the quizzes on each retrieval, but I would think the get request does that automatically.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

With the exception of the QuestionPage.vue everything seemed necessary. I didn't think the code seemed overly complicated, and while there may be some efficiencies to be had, I struggled so much myself, I could not see anything that stood out to me. It seemed to be pretty clean and mostly simplistic.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

I am terrible at adding comments myself, but I do think they can be quite helpful, especially in a case like this where a review process is taking place, but mostly, I would have liked to better understood the point and function of the ```remodelData()``` function.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

The missing .htaccess and .env.production files I think may have been helpful. I believe the functionality still works despite the missing .env.production because the app.js file points to the production domain instead of local domain, but I think that could have unintended consequences when making updates during development.

**Do you have any additional comments not covered in the above questions?**

I like the clean design. It was pleasant to look through. 

I would suggest considering putting more instructions - for instance, distinguishing between the Add Question and Add Test Question buttons. You may also want to reduce the answer minimum to 1 character as I can see how that could be an issue with numeric answers, and I couldn't add the word "No."

It isn't particularly clear that the Question Category is actually the Quiz umbrella, and it would be nice to have a drop-down or something for the images in order to select from them as I was not sure what put. 

Ultimately, it was a fun to play with and seemed to meet the brief.

**Comment by keamarg** [comment]
Hi Charity

First of all, thanks a lot for this. It was extremely helpful, and made me realise, that I had forgotten the .htacces and .env files on the production server, which made the project exhibit the strange behaviour you describe.

All of the things you mention are now addressed (although some of them are just quick-fixes to the bigger design issues I had not thought through before starting the project). 
- Fields are now 1-x characters
- Short explanation in the image field, that the word you are entering is actually a search query for an unsplash image of the topic.
- No 404's.
- Routersystem working with dynamic id-binding (which also means, that the QuestionPage.vue is now being used)
- "Add test data" button was renamed to "Fill with dummy data"

And just to clarify the remodelData() function is there because I could not get the API seeds to accept nested objects, so it is basically taking the question data and creating nested answers as that was how I designed the project in the first place. 

Thanks again! 

Ps: Kudos for juggling it all, I know how stressfull the combination of job+education+family can be. I am so fortunate to be doing these courses "on the clock", but I can imagine how a toddler can throw a wrench in the works (although I must say, I do miss those "cute baby wrenches" sometimes, having an 11-year old in the house;)

**Comment by hhuang56** [comment]
**What is the Github URL of the project you are reviewing?**
[Pandemic Cafe](https://github.com/agt25/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
The author mentioned one in her README about the music breaking if you finish all the tracks. I want to specify that you don't need to iterate through all the tracks to encounter it; just hit play on the first track and then hit the << button to try to go backwards. If you do that, you can't ever get the music going again until you refresh the page.

The "like" button is also somewhat finicky for me. Sometimes it works and sometimes it doesn't. It *might* be that I'm not waiting long enough between clicks for the heart to change from unfilled to filled? But I give it at least 10 seconds before trying again, and some times it really just doesn't update the button until I click twice. Maybe change the appearance of the button immediately but let the async call resolve in the background so the user doesn't sit there spam clicking it thinking it's not working?

Some UI confusion points:
- The L I K E S letters appear clickable on the top of the likes page ({"cursor":"pointer"} styling might be going on), but when I click on them nothing happens.
- The word "instrumental" is partially hidden behind the navbar on the explore page. Just needs some more padding on the bottom of the page to fix.

**Are there any parts of the code that you found interesting or taught you something new?**
I just thought it was really cool you hosted tracks on AWS and used them for this project. Really going above and beyond!

**Are there any parts of the code that you don't understand?**
No, everything is very thoroughly documented.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
There are several instances where options are created but do not return anything. It's better to just delete all these empty options. For example, in AboutSection.vue:

    mounted() {
       
        
    },

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
No, overall very thorough commenting. I actually felt there was a bit too much commenting (ie., commenting could be reduced if pressed for time). For example, explicit method names like nextTrack() and previousTrack() in CoffeePod.vue are more than enough to tell me what they do.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
I recommend deleting all your console logs before building for production. Sometimes you can accidentally log something you really don't want users to see...

**Do you have any additional comments not covered in the above questions?**
Very beautiful and artistic UI! The author definitely put a lot of effort to make the project feel like a complete product.

**Comment by seanbollin** [comment]
**What is the Github URL of the project you are reviewing?**

https://github.com/oms210/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

I get a "cannot read property push of undefined" when I click the add icon next to the ingredients here: http://e28p2.oms210.me/recipe/3

Clicking on an ingredient from the list takes me to a blank page: http://e28p2.oms210.me/shoppinglist/6

**Are there any parts of the code that you found interesting or taught you something new?**

Learned about "baseline" in vertial-align

**Are there any parts of the code that you don't understand?**

No everything seems straightforward.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

No everything seems canonical and straightforward.

There is one file that is missing code: https://github.com/oms210/e28/blob/main/p2/src/components/pages/AddIngredient.vue

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

No everything was clear and I did not see the need for any comments.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

Best practices were used. You can refresh the page and it looks like the .htaccess file is in place. Also scoped styles were used.

**Do you have any additional comments not covered in the above questions?**

I like the look and feel of the site. Great job overall!

**Comment by jaredmimms** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/oms210/e28/tree/main/p2](https://github.com/oms210/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
When I click on an ingredient of the food item I get: TypeError: Cannot read property 'push' of undefined. 

**Are there any parts of the code that you found interesting or taught you something new?**
I found the use of the interactive ingredients as items in a shopping list interesting.

**Are there any parts of the code that you don't understand?**
I understand the code but see non-working sections such as adding an ingredient to a non-existent list on line 90 of ShowRecipe.vue:

`this.ingredientslist.push(ingredientListItem);`

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
The author could explicitly define the ingredient list in ShowRecipe.vue.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
I did not need comments to understand the code.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
The code could explicitly define data structures better to overcome errors. 

**Do you have any additional comments not covered in the above questions?**
The project should have included one more element of functionality to address the project requirements.

**Comment by misskimdo** [comment]
**What is the Github URL of the project you are reviewing?**

[FlashCard Trivia](http://e28p2.trevjburns.me/)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 

Accessing any decks/cards gave me a 404. 

I see in your main.js, you created a Vue router for your main nav, but it wasn't used for the decks. Maybe something like: 
```
<router-link
                v-for="deck in decks"
                v-bind:key="deck.name"
                v-bind:to="'/deck/' + deck.name"
                >
                <div id="deck">
                <show-deck<show-decks v-bind:cards='flashcards' v-bind:decks='decks'></show-decks>
                </div>
        </router-link>
```

**Are there any parts of the code that you found interesting or taught you something new?**

- Curious what [-1](https://github.com/trevjburns/e28/blob/416b69f8d0187186f93a8015eeafe7cddfcf2841/p2/src/components/pages/DeckPage.vue#L51) accomplishes?
- I can't access the actual cards page, but I'm very interested in seeing it!


**Are there any parts of the code that you don't understand?**

- This is just a curiosity on my part, but why a [computed ```currentCard()```](https://github.com/trevjburns/e28/blob/416b69f8d0187186f93a8015eeafe7cddfcf2841/p2/src/components/pages/DeckPage.vue#L65) in this situation?

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

- Just the items I mentioned above.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

I did this also, but I did not leave comments. I see now that some items would be better if comments were available. E.g. The ```currentCard()``` I mentioned above.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

The main thing is resolving the issue of being able to access all the decks and cards.

**Do you have any additional comments not covered in the above questions?**

N/A

**Comment by agt25** [comment]
**What is the Github URL of the project you are reviewing?**
https://github.com/sgreene/e28/tree/main/p2 

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

Unfortunately, it appears that the app is making requests to the creator’s local server. 
`Failed to load resource: net::ERR_NAME_NOT_RESOLVED` followed by: `e28api.trinigooner.loc/task:1` 

I also ran into this bug! I had forgotten to update my e28api url after pushing it to production. It can be so frustrating since your local server is probably running while you’re testing the project before submission; thus, you won’t even notice the bug! 

I do see that Simeon did update his e28api url. In theory, everything should be working, right? I believe it is possible that after you pushed your API to production and updated the url, you may have forgotten to run `npm run build` again? Followed by a `git push` and then `git pull` to synchronize the changes with DigitalOcean? At least, that’s a mistake I committed after updating the API url! 

**Are there any parts of the code that you found interesting or taught you something new?**

I thought the refreshTasks method on App.vue was quite neat! I love how elegant this line is: `this.allTasks.sort((a,b) => (a.priority > b.priority) ? 1:-1)` 

**Are there any parts of the code that you don't understand?**

Nope! Simeon did a great job keeping his methods clear and concise. 

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

Nope! All the code is pretty straightforward. 

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

I am a sucker for comments. So, I always appreciate reading some. But truthfully, I don’t think there was a need for comments here. All of Simeon’s methods are so elegant! 

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

Nope; Simeon followed best practices all around! 

**Do you have any additional comments not covered in the above questions?**

Great job on P2 :)

**Comment by sgreene** [comment]
**What is the Github URL of the project you are reviewing?**
[https://github.com/millierni/e28/tree/main/p2](https://github.com/millierni/e28/tree/main/p2)

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**
I did not see any errors in the console. I have basic knowledge of crypto currencies, and while I'm not familiar with the relationship between all these attributes, I tried out various combos and clicked on links all the while keeping the dev console open and I did not see any errors.

**Are there any parts of the code that you found interesting or taught you something new?**
I thought the use of a watcher to update other properties when a miner was selected was interested.

**Are there any parts of the code that you don't understand?**
No. The code is very straightforward and minimalist. That being said, even though this is a calculator I was not familiar with a lot of the calculations and thus learned some new math related to cryptos!

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**
I thought the code was pretty clear. Some of the numbers in the app are pretty large though. I would suggest either using a different widget for displaying them or a different unit, like "K" for thousands. The large numbers are hard to follow and a little rough on the eyes. At least mine. But this is more a style issue.

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**
Without domain knowledge, it's hard to follow the calculations. So I would suggest a few minor comments explaining the calculator fields and their relationship to each other.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**
No. Best practices were followed throughout.

**Do you have any additional comments not covered in the above questions?**
Crypto is a hot field today and this app seems useful. Good job!

**Comment by nverno** [comment]
**What is the Github URL of the project you are reviewing?**
https://github.com/bryonjustice/e28/tree/main/p2

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?**

No, it all works well!

Are there any parts of the code that you found interesting or taught you something new?

I liked the division of the codebase into Show*.vue components and those in pages that made API requests.  I tried to follow a similar design pattern.

The handling of validation errors is also very clean.

**Are there any parts of the code that you don't understand?**

No

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

I think it would be nice to store the test data in a separate location outside of the actual code.  But, I don't have a specific 
recommendation of alternative format at this point :(

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?((

No, the code and variable names are self-explanatory.  Personally I don't like excessive commenting unless there is
some complicated algorithm to explain, so this was a nice codebase to read through.

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

No, it looks good.

**Do you have any additional comments not covered in the above questions?**

The production site looks very nice!

**Comment by katz2row** [comment]
First off, I just was clicking buttons, and I randomly stopped on a song, and it was Leave the Door Open, which I have heard so many times on TikTok that it will be forever burned in my brain. LOL. So that made me laugh.

**What is the Github URL of the project you are reviewing?**

<http://e28p2.vue28.me/>

**Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console?** 

The only error I noticed is when I click on the back button on the player on the homepage and it is at the beginning, I get a 403 error. 
``` GET https://track09.s3.us-east-2.amazonaws.com/-1.mp3 403 (Forbidden) ```

I did expect there to be items I could further click on within the Likes and Explore or a way to modify the results, but I understand they were probably meant to be static pages for the purposes of the project. I thought if I clicked on a track it would start playing on the Likes page and that I could see more information if I clicked on one of the topics on Explore, but that may have just been too much extra, which is understandable for a school project.

Maybe the iPod could float on the right, and then the clicks on the like page could trigger the iPod to play at that track?

There also seems to be a little bit of UI scrolling issues with the fixed footer when you expand About and "instrumental" on the Explore page was always partially covered.

**Are there any parts of the code that you found interesting or taught you something new?**

I liked the simplistic but elegant way that the About collapse was included. 

I also noticed you are using emits, which has been plaguing me. It seems so simplistic, but I could not for the life of me get it to work. 

**Are there any parts of the code that you don't understand?**

I am honestly not sure what the Explore/Categories page is aiming to do. It doesn't seem to influence the Lounge player, and nothing on the page seems clickable or editable. It didn't seem to change depending on the song I was on or what I liked. Since they were fixed categories from all available songs, I just didn't know if I was supposed to be doing or learning something on that page.

**Are there any parts of the code that you feel could be written in a more efficient or clearer manner?**

I think everything seemed pretty clean. The commenting really helped. I think the references may have better been pointed to @ to navigate to the src directory, but otherwise it looked clean. 

**Are there aspects of the code that you feel were not self-evident and would benefit from comments?**

There were great comments that really helped me. I am terrible at commenting, and sites like this really show their value. 

**Are there any best practices discussed in course material that you feel were not addressed in the code?**

Except for the path to the src directory, it seemed pretty much to have followed the best practices. 

**Do you have any additional comments not covered in the above questions?**

It was very nice. I like the graphics and the lounge table. Nice job! 

The one thing I would consider is the fixed footer because on my laptop screen, it requires a lot of scrolling to access things, and on my larger screen, it still covered parts of the iPod, etc. without scrolling. Maybe, if still fixed, it could be floated on one side or the other with the cups closer in order to prevent the overlap and scrolling. Or maybe a smaller table with a laptop and above it, shelves, each with a navigation cup. Just thoughts. Otherwise, great!

**Comment by dan-harnden** [comment]
What is the Github URL of the project you are reviewing?

[]([](https://github.com/kerynegan/e28/tree/main/p2))

Are you able to produce any errors or unexpected results when using the site? Are there any errors or warnings in the console? 

“DevTools failed to load SourceMap: Could not load content for http://e28p2.kerynegan.me/js/chunk-vendors.fb149b60.js.map: Load canceled due to load timeout”

This did appear after the site was up for a little time. Prior to that not errors were noted.


Are there any parts of the code that you found interesting or taught you something new?

No, it was very literally by the lesson plan structure.


Are there any parts of the code that you don't understand?

No, as stated above it meets standards as taught.


Are there any parts of the code that you feel could be written in a more efficient or clearer manner?

I don’t believe so, that I could note.


Are there aspects of the code that you feel were not self-evident and would benefit from comments?

I think commenting is great, I often over comment. I’ve found in my trade which is IT related, I seldom have to do programming, but have to dissect it at times. I comment everything I learn in them. I notate actions taken to fix issues too. I find it very helpful, as I don’t want to spin my wheels in circles retying things I already tried. So, my advice is to comment often.


Are there any best practices discussed in course material that you feel were not addressed in the code?

No, not at this time.


Do you have any additional comments not covered in the above questions?

Nice job in sticking to the curriculum and working through it.
